lib/node_modules/@stdlib/plot/table/unicode/lib/main.js

```diff
@@ -0,0 +1,1245 @@
+var getMargins = require( './props/margins/get.js' );
+var setMaxColumnWidth = require( './props/max-column-width/set.js' );
+var getMaxColumnWidth = require( './props/max-column-width/get.js' );
+var setMaxWidth = require( './props/max-width/set.js' );
+var getMaxWidth = require( './props/max-width/get.js' );
+var setPaddingLeft = require( './props/padding-left/set.js' );
+var getPaddingLeft = require( './props/padding-left/get.js' );
+var setPaddingRight = require( './props/padding-right/set.js' );
+var getPaddingRight = require( './props/padding-right/get.js' );
+var setRowSeparator = require( './props/row-separator/set.js' );
+var getRowSeparator = require( './props/row-separator/get.js' );
+var setVerticalSeparatorMode = require( './props/vertical-separator-mode/set.js' );
+var getVerticalSeparatorMode = require( './props/vertical-separator-mode/get.js' );
+var defaults = require( './defaults.js' );
+var render = require( './render.js' );
+var push = require( './push.js' );
+var getter = require( './get.js' );
+var setter = require( './set.js' );
+
+
+// VARIABLES //
+
+var debug = logger( 'table:unicode:main' );
+
+// List of private properties (note: keep in alphabetical order):
+var PRIVATE_PROPS = [
+	'_align',
+	'_autoRender',
+	'_borders',
+	'_bufferSize',
+	'_corners',
+	'_columnSeparator',
+	'_columnWidth',
+	'_data',
+	'_format',
+	'_headerAlign',
+	'_headers',
+	'_headerSeparator',
+	'_horizontalSeparatorMode',
+	'_joints',
+	'_margins',
+	'_maxColumnWidth',
+	'_maxWidth',
+	'_numColumns',
+	'_paddingLeft',
+	'_paddingRight',
+	'_quiet',
+	'_rowSeparator',
+	'_verticalSeparatorMode',
+	'_widthUnits'
+];
+
+// List of options properties (note: keep in alphabetical order):
+var OPTIONS_PROPS = [
+	'align',
+	'autoRender',
+	'borderBottom',
+	'borderLeft',
+	'borderRight',
+	'borders',
+	'borderTop',
+	'bufferSize',
+	'columnSeparator',
+	'columnWidth',
+	'cornerBottomLeft',
+	'cornerBottomRight',
+	'corners',
+	'cornerTopLeft',
+	'cornerTopRight',
+	'format',
+	'headerAlign',
+	'headers',
+	'headerSeparator',
+	'horizontalSeparatorMode',
+	'jointBottomLeft',
+	'jointBottomRight',
+	'jointMiddle',
+	'joints',
+	'jointTopLeft',
+	'jointTopRight',
+	'marginBottom',
+	'marginLeft',
+	'marginRight',
+	'marginTop',
+	'margins',
+	'maxColumnWidth',
+	'maxWidth',
+	'paddingLeft',
+	'paddingRight',
+	'rowSeparator',
+	'verticalSeparatorMode'
+];
+
+
+// FUNCTIONS //
+
+/**
+* Initializes private instance properties.
+*
+* @private
+* @param {UnicodeTable} table - table instance
+* @param {Object} defaults - default property values
+* @returns {UnicodeTable} table instance
+*/
+function initializePrivateProperties( table, defaults ) { // eslint-disable-line id-length
+	var i;
+
+	for ( i = 0; i < PRIVATE_PROPS.length; i++ ) {
+		defineProperty( table, PRIVATE_PROPS[ i ], {
+			'configurable': false,
+			'enumerable': false,
+			'writable': true,
+			'value': defaults[ PRIVATE_PROPS[ i ].substring( 1 ) ] || null
```

Comment:
What if default value is `false` in that case it store the `null` value , so i believe logic should be 
```suggestion
			'value': ( defaults[ PRIVATE_PROPS[ i ].substring( 1 ) ] === undefined ) ? null : definedValue
```

Reply:
Yes, I think this is correct. We should account for falsy default values.

Reply:
Although, I think we want to refactor to avoid repeated substring creation. E.g., by assigning to a temporary variable within the loop.

```javascript
v = defaults[ PRIVATE_PROPS[ i ].substring( 1 ) ];

...

'value': ( v === void 0 ) ? null : v
```